swiftide-test-utils/src/test_utils.rs

```diff
@@ -70,6 +73,41 @@ pub async fn start_redis() -> (ContainerAsync<GenericImage>, String) {
     (redis, redis_url)
 }
 
+/// Setup Postgres container.
+/// Returns container server and `server_url`.
+pub async fn start_postgres() -> (ContainerAsync<GenericImage>, String, TempDir) {
+    // Create a temporary directory for Postgres data
+    let temp_dir = TempDir::new().expect("Failed to create temp folder");
+    let temp_data_bind_path = temp_dir.path().to_str().unwrap();
+
+    // Find a free port on the host for Postgres to use
+    let host_port = portpicker::pick_unused_port().expect("No available free port on the host");
+
+    let postgres = testcontainers::GenericImage::new("ankane/pgvector", "v0.5.1")
+        .with_wait_for(WaitFor::message_on_stdout(
+            "database system is ready to accept connections",
+        ))
+        .with_mapped_port(host_port, 5432.tcp())
+        .with_env_var("POSTGRES_USER", "myuser")
+        .with_env_var("POSTGRES_PASSWORD", "mypassword")
+        .with_env_var("POSTGRES_DB", "mydatabase")
+        .with_mount(Mount::bind_mount(
+            temp_data_bind_path,
+            "/var/lib/postgresql/data",
+        ))
```

Comment:
Given the resource constraints in the CI/CD build context, `Mount::tmpfs_mount()` seems like a promising choice since the volume resource is released once the container is stopped.

```rust
let postgres = testcontainers::GenericImage::new("pgvector/pgvector", "pg17")
    .with_wait_for(WaitFor::message_on_stdout(
        "database system is ready to accept connections",
    ))
    .with_mapped_port(host_port, 5432.tcp())
    .with_env_var("POSTGRES_USER", "myuser")
    .with_env_var("POSTGRES_PASSWORD", "mypassword")
    .with_env_var("POSTGRES_DB", "mydatabase")
    .with_mount(Mount::tmpfs_mount("/var/lib/postgresql/data"))
    .start()
    .await
    .expect("Failed to start Postgres container");
```

Does this approach fit the context, or would you suggest any adjustments?